mkdocs.yml

```diff
@@ -14,6 +14,7 @@ nav:
           - Resource: add-a-new-resource.md
           - Service: add-a-new-service.md
           - Data source: add-a-new-datasource.md
+          - Function: add-a-new-function.md
```

Comment:
Do we need a little more than just "Function"? Provider function? Not sure.

Reply:
I went with `Function` to align with how the Terraform Plugin Framework package is named, and how its used in the documentation:
https://developer.hashicorp.com/terraform/plugin/framework/functions

"Provider-defined" function is another term that has been used, and though I think the location of this document (the contributing docs for a provider) implies that namespace, I don't have a strong opinion either way.